frontend/src/metabase/reference/metrics/MetricListContainer.jsx

```diff
@@ -0,0 +1,72 @@
+/* eslint "react/prop-types": "warn" */
+import React, { Component } from 'react';
+import PropTypes from "prop-types";
+import { connect } from 'react-redux';
+
+import BaseSidebar from 'metabase/reference/guide/BaseSidebar.jsx';
+import SidebarLayout from 'metabase/components/SidebarLayout.jsx';
+import MetricList from "metabase/reference/metrics/MetricList.jsx"
+
+import * as metadataActions from 'metabase/redux/metadata';
+import * as actions from 'metabase/reference/reference';
+
+import {
+    getDatabaseId,
+    getIsEditing
+} from '../selectors';
+
+const mapStateToProps = (state, props) => ({
+    databaseId: getDatabaseId(state, props),
+    isEditing: getIsEditing(state, props)
+});
+
+const mapDispatchToProps = {
+    ...metadataActions,
+    ...actions
+};
+
+@connect(mapStateToProps, mapDispatchToProps)
+export default class MetricListContainer extends Component {
+    static propTypes = {
+        params: PropTypes.object.isRequired,
+        location: PropTypes.object.isRequired,
+        isEditing: PropTypes.bool
+    };
+
+
+    async fetchContainerData(){
+        await actions.rFetchMetrics(this.props);
+    }
+
+    async componentWillMount() {
+        this.fetchContainerData()
+    }
+
+
+    async componentWillReceiveProps(newProps) {
+        if (this.props.location.pathname === newProps.location.pathname) {
+            return;
+        }
+
+        newProps.endEditing();
```

Comment:
theis action call cluster really, really needs to be in a single action as it is being repeated in many components